stl/inc/yvals_core.h

```diff
@@ -2034,5 +2043,24 @@ compiler option, or define _ALLOW_RTCc_IN_STL to suppress this error.
 #define _RESTRICT __restrict
 #endif // ^^^ !defined(__CUDACC__) ^^^
 
+#if _HAS_CXX20 // see GH-5225
+#if defined(__EDG__)
+#define _HAS_IF_CONSTEVAL _HAS_CXX23
+#elif defined(__clang__) || !defined(__CUDACC__) // ^^^ EDG / Clang or MSVC vvv
+#define _HAS_IF_CONSTEVAL 1
+#else // ^^^ Clang or MSVC / others vvv
+#define _HAS_IF_CONSTEVAL defined(__cpp_if_consteval)
+#endif // ^^^ others ^^^
+
+#if _HAS_IF_CONSTEVAL
+#define _STL_CONSTEVAL_CONDITION consteval
+#else // ^^^ no workaround / workaround vvv
+#define _STL_CONSTEVAL_CONDITION (_STD is_constant_evaluated())

```

Comment:
Unless I am missing something, I believe that we should be able to use this macro down to C++14 mode and have it expand to use `_Is_constant_evaluated` before C++20. Then we can use it consistently throughout the STL.